packages/shopify-app-remix/src/server/authenticate/public/__tests__/factory.test.ts

```diff
@@ -0,0 +1,164 @@
+import {shopifyApp} from '../../..';
+import {
+  APP_URL,
+  getJwt,
+  getThrownResponse,
+  testConfig,
+} from '../../../__test-helpers';
+
+describe('JWT validation', () => {
+  it.only('logs that the method is deprecated', async () => {
+    // GIVEN
+    const config = testConfig();
+    const shopify = shopifyApp(config);
+    const {token} = getJwt();
+
+    // WHEN
+    await shopify.authenticate.public(
+      new Request(APP_URL, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }),
+    );
+
+    // THEN
+    expect(config.logger?.log).toHaveBeenCalledWith(
+      expect.any(Number),
+      expect.stringContaining(
+        'authenticate.public() will be deprecated in v2. Use authenticate.public.checkout() instead.',
+      ),
+    );
+  });
+
+  it('returns token when successful', async () => {
+    // GIVEN
+    const shopify = shopifyApp(testConfig());
+    const {token, payload} = getJwt();
+
+    // WHEN
+    const {sessionToken} = await shopify.authenticate.public(
+      new Request(APP_URL, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }),
+    );
+
+    // THEN
+    expect(sessionToken).toMatchObject(payload);
+  });
+
+  it('sets extra CORS allowed headers when requested from a different origin', async () => {
+    // GIVEN
+    const shopify = shopifyApp(testConfig());
+    const {token} = getJwt();
+
+    // WHEN
+    const {cors} = await shopify.authenticate.public(
+      new Request('https://some-other.origin', {
```

Comment:
To be super pedantic, we should probably use `APP_URL` here to make it a "proper" CORS request. Irrelevant as far as the code goes, but might make the test easier to understand.

Reply:
Yeah, fair. Updated 👍